File: Practice/Bcontest129_b.cpp
#include <bits/stdc++.h>
using namespace std;

#define rep(i, n) for (int i = 0; i < (int)(n); i++)

int main() {
    long long N, ans = 1000000007;
    cin >> N;
    vector<long long> W(N), W_sum(N+1,0);
    rep(i, N){
        cin >> W[i];
    }

    // 累積和の作成
    rep(i,N) {
        W_sum[i+1] = W_sum[i] + W[i];
    }


    for(int i = 1; i <= N+1; ++i) {
        ans = min(ans, abs((W_sum[N] - W_sum[i]) - W_sum[i])); 
    }

    cout << ans << endl;
    return 0;
}

File: Practice/ABC/Bcontest129_b.py
N = int(input())
W = list(map(int,input().split()))
# abs_list = list()
# for i in range(len(W)):
# 	abs_list.append(abs(sum(W[:i+1])-sum(W[i+1:])))
# print(min(abs_list))

W_sum = list()
W_sum.append(0)
for i in range(N):
	W_sum.append(W_sum[i] + W[i])

ans = 100000007
for i in range(len(W_sum)):
	ans = min(ans, abs((W_sum[N] - W_sum[i]) - W_sum[i]))

print(ans)
